src/config.c

```diff
@@ -1031,6 +1063,14 @@ struct rewriteConfigState *rewriteConfigReadOldFile(char *path) {
             continue;
         }
 
+        /* If we have a module config in the config file, but its no longer present: remove it. */
+        if (strchr(argv[0], '.') && !isConfigNameRegistered(argv[0])) {
```

Comment:
The situation that this accounts for is 1. A module load with configs. 2. a rewrite. 3. a module unload, and then 4. another config rewrite. This causes the server to not start-up from the .conf file because the module configurations are still in the .conf file.

Reply:
The server will fail startup after stage **2** in the above description (in case it doesn't start with the module). Just the fact that we write module confs into the config file has the potential to cause a startup issue. But we still want to do so.
So my suggestion is either
1.  Allow this, and in case of a startup issue the user will need to fix their config file.
2. **Any** unrecognized config will be excluded during the rewrite (no matter if it's a module or not). Or even better commented out during the rewrite.
3. Extend module unload command to optionally rewrite the config during unload if we've performed a rewrite since the load. In this rewrite we explicitly skip this module's configs.

(I'm for 2 with the "comment" extension)